File: scripts/test_api.mjs
// Quick API fetch test
async function run() {
    const url = 'http://localhost:3000/api/aws/behavior?kd=3155&start=2026-01-02&end=2026-03-21';
    console.log('Fetching:', url);
    try {
        const res = await fetch(url);
        const data = await res.json();
        console.log('Response status:', res.status);
        if (data.roster) {
            console.log('Roster length:', data.roster.length);
            if (data.roster.length > 0) {
                console.log('Sample profile:', JSON.stringify(data.roster[0], null, 2));
            }
        } else {
            console.log('Data:', data);
        }
    } catch(e) {
        console.error(e);
    }
}
run();

File: scripts/test_api.py
import urllib.request
import json

ports = [3000, 3001, 8080]
endpoint = "/api/aws/behavior?kd=3155&start=2026-01-02&end=2026-03-21"

success = False
for port in ports:
    url = f"http://localhost:{port}{endpoint}"
    print(f"Trying {url}...")
    try:
        req = urllib.request.Request(url)
        with urllib.request.urlopen(req, timeout=5) as response:
            data = json.loads(response.read().decode())
            print(f"✅ Success on port {port}!")
            roster = data.get("roster", [])
            print(f"Returned {len(roster)} governors.")
            if len(roster) > 0:
                print("Sample keys:", list(roster[0].keys()))
            success = True
            break
    except Exception as e:
        print(f"❌ Failed: {e}")

if not success:
    print("\nCould not connect to the dev server. It may be offline or on an unknown port.")
